src/make_trade.ts

```diff
@@ -74,46 +81,47 @@ export async function makeTrade(
     ethers
   );
 
-  const fee = await api.getFee(
+  const { feeAmount, sellAmount, buyAmount } = await getQuote(
+    api,
     sellToken.address,
     buyToken.address,
     sellBalance,
-    OrderKind.SELL
+    trader.address
   );
 
   // This should rarely happen as we only select buy tokens for which fee was sufficient
   // in the first place. Only if approval took a long time and gas prices increased significantly
   // this could be an issue.
-  if (sellBalance.lte(fee)) {
+  if (sellBalance.lte(feeAmount)) {
     throw new Error("Account doesn't have enough balance to pay fee");
   }
 
-  const sellAmountAfterFee = sellBalance.sub(fee);
-  const buyAmount = await api.estimateTradeAmount(
-    sellToken.address,
-    buyToken.address,
-    sellAmountAfterFee,
-    OrderKind.SELL
-  );
-
-  const prettySellAmount = formatAmount(sellAmountAfterFee, sellToken);
+  const prettySellAmount = formatAmount(sellAmount, sellToken);
   const prettyBuyAmount = formatAmount(buyAmount, buyToken);
-  const prettyFee = formatAmount(fee, sellToken);
+  const prettyFee = formatAmount(feeAmount, sellToken);
   console.log(
     `🤹 Selling ${prettySellAmount} of ${sellToken.name} for ${prettyBuyAmount} of ${buyToken.name} with a ${prettyFee} fee`
   );
 
   const order = createOrder(
```

Comment:
Why do we need to create an order here again if `api.getFee` is already returning an order? We could reuse it in theory right?

Just FYI, I'm 100% fine with how it is, just more of a theoretical question.

Reply:
It was only to apply slippage. Will remove.